test/e2e/aws_test.go

```diff
@@ -184,4 +185,22 @@ func TestAWS(t *testing.T) {
 		}
 	}
 
+	// Check that the aws access key and secret key in the system configuration
+	// can be overridden by a user defined secret
+	fakeAWSCredsData := map[string]string{
+		"REGISTRY_STORAGE_S3_ACCESSKEY": "myAccessKey",
+		"REGISTRY_STORAGE_S3_SECRETKEY": "mySecretKey",
+	}
+
+	if err := util.CreateOrUpdateSecret("image-registry-private-configuration-user", "openshift-image-registry", fakeAWSCredsData); err != nil {
+		t.Errorf("unable to create secret \"openshift-image-registry/image-registry-configuration-user\": %#v", err)
+	}
+
+	cfgUser, err := clusterconfig.GetAWSConfig()
+	if err != nil {
+		t.Errorf("unable to get aws configuration: %#v", err)
+	}
+	if fakeAWSCredsData["REGISTRY_STORAGE_S3_ACCESSKEY"] != cfgUser.Storage.S3.AccessKey || fakeAWSCredsData["REGISTRY_STORAGE_S3_SECRETKEY"] != cfgUser.Storage.S3.SecretKey {
+		t.Errorf("expected system configuration to be overridden by the user configuration but it wasn't.")
+	}
```

Comment:
since you've gotten this far, you might as well add a test that confirms we report a useful status condition when you can't create the s3 bucket because the credentials are invalid.

Reply:
updated in current push